apis/v1alpha1/route_types.go

```diff
@@ -19,6 +19,38 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GatewayAllowType specifies which Gateways should be allowed to use a Route.
+type GatewayAllowType string
+
+const (
+	// GatewayAllowAll indicates that all Gateways will be able to use this
+	// route.
+	GatewayAllowAll GatewayAllowType = "All"
+	// GatewayAllowList indicates that only Gateways that have been specified
+	// in GatewayRefs will be able to use this route.
+	GatewayAllowList GatewayAllowType = "List"
```

Comment:
FromList (list is commonly used as a verb in k8s)

Seems pretty intuitive:

```
allow: All
allow: FromList
allow: SameNamespace 
```

Reply:
Updated, thanks!